corehq/form_processor/models.py

```diff
@@ -552,19 +552,43 @@ def xml_md5(self):
         return self.get_attachment_meta('form.xml').content_md5()
 
     def archive(self, user_id=None):
+        # If this archive was initiated by a user, delete all other stubs for this action so that this action
+        # isn't overridden
+        from couchforms.models import UnfinishedArchiveStub
+        UnfinishedArchiveStub.objects.filter(xform_id=self.form_id).all().delete()
```

Comment:
I think this should happen after the check for `self.is_archived` as you have it in the couch version. Same below.

Reply:
https://github.com/dimagi/commcare-hq/pull/22470/commits/2c517d66b377e2c5d2147d56f4fa9c20d5027558